examples/sparse_tensor.cu

```diff
@@ -149,5 +149,17 @@ int main([[maybe_unused]] int argc, [[maybe_unused]] char **argv)
   (X = solve(Acsr, Y)).run(exec);
   print(X);
 
+  //
+  // A direct dense2sparse conversion.
+  //
+  auto D = make_tensor<float, 2>({4, 8});
+  D.SetVals({
+    {0, 11,  0, 12,  0,  0,  0,  0},
+    {0,  0, 13,  0,  0,  0,  0,  0},
+    {0,  0,  0,  0,  0,  0,  0, 14},
+    {0, 15,  0,  0, 16,  0, 17,  0}});
+  (Acoo = dense2sparse(D)).run(exec);
```

Comment:
I would put a comment here with the caveats we discussed. Namely it's only possible with a direct assignment, can be slow, etc